core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -103,7 +103,12 @@
      * in Java {@code \w} is equivalent to {@code [A-Za-z0-9_]} (take care of "_")
      */
     private static final String DEFAULT_ID_REGEX = "^[\\w-]+$";
-    
+
+    /**
+     * Magic header used to detect if a password is bcrypt hashed.
+     */
+    private static final String JBCRYPT_HEADER = "#jbcrypt:";
```

Comment:
FTR that constant was in the `PASSWORD_ENCODER` anonymous class before. There is a goal behind that move.

Reply:
Seems like a bad idea to duplicate this string. More than that, it seems to suggest that these changes are in the wrong location. Might be better to isolate some of this into the `PASSWORD-ENCODER`.